src/test/java/org/minions/devfund/mauricioz/linkedList/RemoveMaximumValuesTest.java

```diff
@@ -0,0 +1,186 @@
+    }
+
+    /**
+     * test.
+     */
+    @Test
+    public void testNull() {
+        try {
+            LinkedListUtils.removeMaximumValues((LinkedList)null, 2);
+        } catch (Exception var2) {
+            Assert.fail("removeMaximumValues throws " + var2 + " when input LinkedList is null");
+        }
+
+    }
+
+    /**
+     * test.
+     */
+    @Test
+    public void testNegativeSize() {
+        LinkedList var1 = new LinkedList();
+        var1.add("doge");
+        var1.add("cat");
+
+        try {
+            LinkedListUtils.removeMaximumValues(var1, -1);
+        } catch (Exception var3) {
+            Assert.fail("removeMaximumValues throws " + var3 + " when number of elements to remove is negative");
+        }
+
+        if (var1.size() < 2) {
+            Assert.fail("removeMaximumValues removes elements from LinkedList when number of elenents to remove is negative");
+        }
+
+        if (var1.size() > 2) {
+            Assert.fail("removeMaximumValues adds elements to LinkedList when number of elenents to remove is negative");
+        }
+
+    }
+
+    /**
+     * test.
+     */
+    @Test
+    public void testInsertEmpty() {
+        LinkedList var1 = new LinkedList();
+
+        try {
+            LinkedListUtils.removeMaximumValues(var1, 1);
+        } catch (Exception var3) {
+            Assert.fail("removeMaximumValues throws " + var3 + " when input LinkedList is empty");
+        }
+
+        Assert.assertTrue("removeMaximumValues adds elements to list when input LinkedList is empty", var1.isEmpty());
+    }
+
+    /**
+     * test.
+     */
+    @Test
+    public void testRemoveGreaterThanSize() {
+        LinkedList var1 = new LinkedList();
+        var1.add("doge");
+
+        try {
+            LinkedListUtils.removeMaximumValues(var1, 2);
+        } catch (Exception var3) {
+            Assert.fail("removeMaximumValues throws " + var3 + " when number of elements to remove is greater than size of input LinkedList");
+        }
+
+        Assert.assertTrue("removeMaximumValues does not remove all elements when number of elements to remove is greater than size of input LinkedList", var1.isEmpty());
+    }
+
+    /**
+     * test.
+     */
+    @Test
+    public void testRemoveEqualsSize() {
+        LinkedList var1 = new LinkedList();
+        var1.add("doge");
+
+        try {
+            LinkedListUtils.removeMaximumValues(var1, 1);
+        } catch (Exception var3) {
+            Assert.fail("removeMaximumValues throws " + var3 + " when number of elements to remove equals size of input LinkedList");
+        }
+
+        Assert.assertTrue("removeMaximumValues does not remove all elements when number of elements to remove equals size of input LinkedList", var1.isEmpty());
+    }
+
+    /**
+     * test.
+     */
+    @Test
+    public void testEachMaxAppearsOnce() {
+        LinkedList var1 = new LinkedList();
+        var1.add("doge");
+        var1.add("cat");
+        var1.add("kangaroo");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define a constant instead of duplicating this literal "kangaroo" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)